core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -367,8 +368,8 @@ public Response getWebUIBrowse(@DefaultValue("/") @QueryParam("path") String req
           ServerConfiguration.getBoolean(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_ENABLED))
           .setMasterNodeAddress(mMasterProcess.getRpcAddress().toString()).setInvalidPathError("");
       List<FileInfo> filesInfo;
-      String path = requestPath;
-      if (path == null || path.isEmpty()) {
+      String path = URLDecoder.decode(requestPath, "UTF-8");
+      if (path.isEmpty()) {
         path = AlluxioURI.SEPARATOR;
```

Comment:
don't we need to encode this too?

Reply:
We only need to decode what is sent to us by the client in `requestPath`.